### DISCUSSION TITLE
How to add a computing latency in a piodevice

### DISCUSSION THREAD

**Discussion by dreamerliii** [gem5/gem5#978] [Q&A] [answered]
Hi, I'm tring to use a piodevice to do a job, such as 'int n = 1+1'. I find that this step use 0 tick, but actually it should need a latency to finish an "add", how can I add that latency ? I need to see this latency in the m5out/stats.txt. Thanks !

**Comment by robhau** [comment, marked as answer]
In a `BasicPioDevice`, you can set `pio_latency`.

https://github.com/gem5/gem5/blob/ee6f1377d7c54422137dfa47cd4d73407814867d/src/dev/Device.py#L83-L90

After accepting a request, the answer be will sent after this latency.

See also:
https://github.com/gem5/gem5/blob/ee6f1377d7c54422137dfa47cd4d73407814867d/src/mem/tport.cc#L62-L86


Therefore, you have to increase this latency by the time that is needed for the addition.
